include/seqan3/search/fm_index/fm_index.hpp

```diff
@@ -235,10 +278,10 @@ class fm_index
         // uint8_t largest_char = 0;
         sdsl::int_vector<8> tmp_text(text.size());
 
-        //TODO view::reverse is broken and can't be chained right now
-        std::vector<value_type_t<text_t>> another_copy = text | view::reverse;
-
-        std::ranges::copy(another_copy | view::to_rank | view::transform([] (uint8_t const r) { return r + 1; }),
+        std::ranges::copy(text
+                          | view::to_rank
+                          | view::transform([] (uint8_t const r) { return r + 1; })
```

Comment:
I would instead insert into the lambda:
```cpp
| view::transform([] (uint8_t const r)
{
    if constexpr (alphabet_size_v<char_type> >= 256 - is_collection)
        if (r >= 256 - 1- is_collection)
            throw std::out_of_range("The input text cannot be indexed, because for full character alphabets the last one/two values are reserved (single sequence/collection).");
    return r + 1;
}
```

This way you make sure that the check is performed in release mode, too, but that you also only perform it if the underlying alphabet could possibly have such high values.